src/user/MOM_wave_interface.F90

```diff
@@ -1537,6 +1746,51 @@ subroutine Waves_end(CS)
 
 end subroutine Waves_end
 
+!> Register wave restart fields. To be called before MOM_wave_interface_init
+subroutine waves_register_restarts(CS, HI, GV, param_file, restart_CSp)
+  type(wave_parameters_CS), pointer       :: CS           !< Wave parameter Control structure
+  type(hor_index_type),     intent(inout) :: HI           !< Grid structure
+  type(verticalGrid_type),  intent(in)    :: GV           !< Vertical grid structure
+  type(param_file_type),    intent(in)    :: param_file   !< Input parameter structure
+  type(MOM_restart_CS),     pointer       :: restart_CSp  !< Restart structure, data intent(inout)
+  ! Local variables
+  type(vardesc) :: vd(2)
+  logical :: use_waves
+  logical :: StatisticalWaves
+  character*(13) :: wave_method_str
+  character(len=40)  :: mdl = "MOM_wave_interface" !< This module's name.
+
+  if (associated(CS)) then
+    call MOM_error(FATAL, "waves_register_restarts: Called with initialized waves control structure")
+  endif
+  allocate(CS)
+
+  call get_param(param_file, mdl, "USE_WAVES", use_waves, &
+       "If true, enables surface wave modules.", do_not_log=.true., default=.false.)
+
+  ! Check if using LA_LI2016
+  call get_param(param_file,mdl,"USE_LA_LI2016",StatisticalWaves,     &
+                 do_not_log=.true.,default=.false.)
+
+  if (.not.(use_waves .or. StatisticalWaves)) return
+
+  ! Allocate wave fields needed for restart file
+  allocate(CS%Us_x(HI%isdB:HI%IedB,HI%jsd:HI%jed,GV%ke))
+  CS%Us_x(:,:,:) = 0.0
+  allocate(CS%Us_y(HI%isd:HI%Ied,HI%jsdB:HI%jedB,GV%ke))
+  CS%Us_y(:,:,:) = 0.0
+
+  call get_param(param_file,mdl,"WAVE_METHOD",wave_method_str, do_not_log=.true., default=NULL_STRING)
+
+  if (trim(wave_method_str)== trim(SURFBANDS_STRING)) then
+    vd(1) = var_desc("US_x", "m s-1", "3d zonal Stokes drift profile")
+    vd(2) = var_desc("US_y", "m s-1", "3d meridional Stokes drift profile")
+    call register_restart_field(CS%US_x(:,:,:), vd(1), .true., restart_CSp)
```

Comment:
The logical here has to be set to .false. to work in symmetric memory mode.  I do not understand why.